File: JavaCivitas/src/civitas/CasillaSorpresa.java
package civitas;

import java.util.ArrayList;

public class CasillaSorpresa extends Casilla{

    private MazoSorpresas mazo;

    CasillaSorpresa(MazoSorpresas mazo_a, String nom){

        super(nom);
        mazo = mazo_a;

    }

    @Override
    public void recibeJugador(int iactual, ArrayList<Jugador> todos){

        if(jugadorCorrecto(iactual, todos)){
            Sorpresa sorpresa = mazo.siguiente();
            sorpresa.aplicarAJugador(iactual,todos);
            super.informe(iactual,todos);
        }

    }

}

File: civitasruby/pkg/Civitas-0.0.1/lib/casilla_sorpresa.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Civitas
class CasillaSorpresa < Casilla
  def initialize(nombre,mazo)
    super(nombre)
    @mazo = mazo
  end
  
  def recibe_jugador(actual,todos)
    if(jugador_correcto(actual, todos))
      super(actual,todos)
      sorpresa1=@mazo.siguiente
      sorpresa1.aplicar_a_jugador(actual, todos)
    end
  end
   
  def to_string
    cadena="Casilla: #{@nombre} \n"
    cadena+="\n"
    return cadena
  end
   public :to_string
end
end
